src/utils/serialize-element.js

```diff
@@ -0,0 +1,194 @@
+// https://stackoverflow.com/questions/6209161/extract-the-current-dom-and-print-it-as-a-string-with-styles-intact
```

Comment:
I don't fully understand how the SVG hack works, will we still need this if we render the react component in a way where it can easily toggle its own styles?

Reply:
the intention here was to make it so the internal styles and structure of the chat messages will be future proofed -- for example, if we were to add markdown support, it would "just work" without altering the SVG generation code. so this bit of code basically just inlines all the styles in a way that is agnostic to our overall CSS structure.

i realize I'm slightly breaking this contract right now by having some light styling applied by the wrapper div, but it still seems like a good design to make the inner parts of the chat message and the CSS styles be 1-1 against whatever we would render in the real DOM for the chat history. the current emoji library and blockquote functionality illustrate the versatility somewhat -- there was no new code needed in the SVG generation to pull that functionality in.